src/lib.rs

```diff
@@ -9,30 +9,54 @@ use std::io::Write;
 use std::process::{Command, Stdio};
 
 /// Describes the style to pass to clang-format
+///
+/// This list is created from
+/// <https://clang.llvm.org/docs/ClangFormatStyleOptions.html#basedonstyle>
 #[derive(Debug, PartialEq)]
 pub enum ClangFormatStyle {
+    /// A style complying with [Chromium’s style guide](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/styleguide/styleguide.md)
     Chromium,
+    /// Use the default clang-format style
     Default,
+    /// clang-format will try to find the .clang-format file located in the closest parent directory of the current directory.
     File,
+    /// A style complying with the [GNU coding standards](https://www.gnu.org/prep/standards/standards.html)
+    ///
+    /// Since clang-format 11
+    GNU,
+    /// A style complying with [Google’s C++ style guide](https://google.github.io/styleguide/cppguide.html)
     Google,
+    /// A style complying with the [LLVM coding standards](https://llvm.org/docs/CodingStandards.html)
     Llvm,
+    /// A style complying with [Microsoft’s style guide](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference)
+    ///
+    /// Since clang-format 9
+    Microsoft,
+    /// A style complying with [Mozilla’s style guide](https://firefox-source-docs.mozilla.org/code-quality/coding-style/index.html)
     Mozilla,
+    /// A style complying with [WebKit’s style guide](https://www.webkit.org/coding/coding-style.html)
     WebKit,
+    /// Specify a custom input to the `--style` argument of clang-format
+    Custom(String),
 }
 
 impl ClangFormatStyle {
     /// Converts the enum ClangFormatStyle to a string that clang-format expects
-    fn as_str(&self) -> &'static str {
+    fn as_str(&self) -> &str {
```

Comment:
@vimpostor @LeonMatthesKDAB do we think that this is an API break ? So we need to bump to `0.3` rather than `0.2.1` ?

Reply:
And while we are changing this do we think that `&str` is best here is would eg `String` be better ? As this could then allow us to use things like `format!` in the future (if ever required for some reason).

Reply:
Do note however, that simply adding options to a previously exhaustive enum is a breaking change.
As existing `match` statements will break.

Reply:
Ah yes good point :-)